### ISSUE TITLE
Handle images that contain duplicate sha256 layers

### ISSUE THREAD

**Issue by pkar** [bazel-contrib/rules_oci#257]
I've run into an image that contains duplicate sha256 layers

```
docker pull index.docker.io/hasura/graphql-engine@sha256:aa508dfa1a887684e3fbf4e045084d6fad8257f565bc99827d4b90dcd5a0254a
docker inspect index.docker.io/hasura/graphql-engine@sha256:aa508dfa1a887684e3fbf4e045084d6fad8257f565bc99827d4b90dcd5a0254a

         "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:6021993d84a2d9db89ec74cf44366ae3f2d9f9810f3c77b2bd0cd5ddf14825b1",
                "sha256:e77e0dfc5a653d156055f7262ee9b3202dc47bf3728e38b646ea6c0bea19011f",
                "sha256:1755f0a6c086ce4481af6659aa8c7e5bc50fefbc11ec0659435da60856882959",
                "sha256:b76a02be9f0ebae748887ae7d80faff1b7926fa280c741c1ca0f61cbb0cc3f2d",
                "sha256:38e333f2268b36e11b93daff5ff97782443207ac2068e984fb3bd21172df91b9",
                "sha256:adb0aea3f4ff0fadf90af1a72eeb792b3876afef90eeb4687dc94431bd66eb3f",
                "sha256:5147195cfdac8d087302f1ead5a37f7f74d2691c8a36f4e947e5ca089dd634b0",
                "sha256:d118ddf4315de2c7b453811611f88ae2305bea2875bd93f6596a3489247a6223",
                "sha256:f4a44b248adc792f8844b723a82c1390fcb89bdda3e522fd6d2a3088135105c7",
                "sha256:625e5cec356a004f84203d116e630239caab035f623b973b5653308e71704a27",
                "sha256:36bccf6326a6152eda33902df11d56adefd3b143c81302e35505bd03aff11ab3",
                "sha256:f0badb0d0d5f2cac1deed636936ecadbde56edab98e01c290990596133bf6fc3",
                "sha256:f0badb0d0d5f2cac1deed636936ecadbde56edab98e01c290990596133bf6fc3"
            ]
        },
```

```
# WORKSPACE
oci_pull(
    name = "hasura_graphql",
    # tag = "v2.22.0.cli-migrations-v3.ubuntu",
    digest = "sha256:aa508dfa1a887684e3fbf4e045084d6fad8257f565bc99827d4b90dcd5a0254a",
    image = "index.docker.io/hasura/graphql-engine",
 )
```

```
# BUILD.bazel
load("@rules_oci//oci:defs.bzl", "oci_image")

oci_image(
    name = "oci_image",
    base = "@hasura_graphql",
    tars = [],
    entrypoint = [],
)
```

Which leads to 

```
$ bazel build //:oci_image
ERROR: /Users/pkar/.cache/asbazel/output/external/hasura_graphql_single/BUILD.bazel:43:18: Label '@hasura_graphql_single//:7e74cee710158ba3d1c4c71b7b867b1da2be192c04e2ec83a647aaa39618cc1d' is duplicated in the 'srcs' attribute of rule 'blobs'
ERROR: /Users/pkar/.cache/asbazel/output/external/hasura_graphql/BUILD.bazel:1:6: Target '@hasura_graphql_single//:hasura_graphql_single' contains an error and its package is in error and referenced by '@hasura_graphql//:hasura_graphql'
ERROR: Analysis of target '//:oci_image' failed; build aborted: 
```

Patching https://github.com/bazel-contrib/rules_oci/blob/main/oci/private/pull.bzl#L389 with something like
```
if hash not in tars:
    tars.append(hash)
```
leads to errors further down during https://github.com/bazel-contrib/rules_oci/blob/main/oci/private/pull.bzl#L316 `copy_to_directory` 

I've gotten around this by first flattening the image and uploading to another registry, however this isn't ideal and I was wondering if maybe there is a better workaround?

If it matters this is in macOS on an m1 processor.

**Comment by pkar** [comment]
```
# WORKSPACE
load("@rules_oci//oci:pull.bzl", "oci_pull")

oci_pull(
        name = "hasura",
        # tag = "v2.22.0.cli-migrations-v3.ubuntu",
        digest = "sha256:aa508dfa1a887684e3fbf4e045084d6fad8257f565bc99827d4b90dcd5a0254a",
        image = "index.docker.io/hasura/graphql-engine",
)
```


```
# BUILD.bazel
load("@rules_oci//oci:defs.bzl", "oci_image", "oci_push", "oci_tarball")
load("@aspect_bazel_lib//lib:transitions.bzl", "platform_transition_filegroup")

platform(
    name = "linux_amd64",
    constraint_values = [
        "@platforms//os:linux",
        "@platforms//cpu:x86_64",
        "@io_bazel_rules_go//go/toolchain:cgo_off",
    ],
)

oci_image(
    name = "oci_image",
    base = "@hasura",
    tars = [],
    entrypoint = [],
)

platform_transition_filegroup(
    name = "oci_image_linux_amd64",
    srcs = [":oci_image"],
    target_platform = ":linux_amd64",
)

local_image = "abcd:latest"

oci_tarball(
    name = "oci_tarball_linux_amd64",
    image = ":oci_image_linux_amd64",
    repo_tags = [local_image],
)
```

```
bazel build //:oci_tarball_linux_amd64
cp: /var/folders/8d/107p3fm10dv8_cd80n6y4tqr0000gn/T/tmp.jG86rjXE/blobs/sha256/7e74cee710158ba3d1c4c71b7b867b1da2be192c04e2ec83a647aaa39618cc1d.tar.gz: Permission denied
Target //build_tools/repositories:oci_tarball_linux_amd64 failed to build
Use --verbose_failures to see the command lines of failed build steps.

```

**Comment by thesayyn** [comment]
fixed by #256

**Comment by thesayyn** [comment]
thanks @uhlajs
